### ISSUE TITLE
[Elasticsearch] Combine SearchCondition and EqualCondition resulting in erroneous ELS request

### ISSUE THREAD

**Issue by dhirtzbruch** [schranz-search/schranz-search#313]
I am not sure if this might be an Elastic version or some other issue. We are currently using ELS 7.0, but I setup a blank ELS 8.0 too, which resultet in the same issue.

The following code works perfect:

```php
        $raw = $this->engine->createSearchBuilder()
                            ->addIndex(self::INDEX_NAME)
                            ->addFilter(new SearchCondition($search))
                            ->limit($limit)
                            ->offset($offset)
                            ->getResult();
```

Resulting JSON body:

```json
{
    sort: [ ],
    query: {
        query_string: {
            query: "TES*"
        }
    }
}
```

Changing the search configuration to the following code also works:

```php
        $raw = $this->engine->createSearchBuilder()
                            ->addIndex(self::INDEX_NAME)
                            ->addFilter(new EqualCondition('branch', $branch->getShortname()))
                            ->getResult();
```

Resulting JSON body:

```json
{
    sort: [ ],
    query: {
        bool: {
            filter: [
                {
                    term: {
                        branch.raw: {
                        value: "DE"
                        }
                    }
                }
            ]
        }
    }
}
```

However, combining both:

```php
        $raw = $this->engine->createSearchBuilder()
                            ->addIndex(self::INDEX_NAME)
                            ->addFilter(new SearchCondition($search))
                            ->addFilter(new EqualCondition('branch', $branch->getShortname()))
                            ->getResult();
```

results in the following JSON body:

```json
{
    sort: [ ],
        query: {
        query_string: {
            query: "TES*"
        },
        bool: {
            filter: [
                {
                    term: {
                        branch.raw: {
                            value: "DE"
                        }
                    }
                }
            ]
        }
    }
}
```

which at least both available ELS instances (7.0 and 8.0) don't seem to like...

```json
{"error":{"root_cause":[{"type":"parsing_exception","reason":"[query_string] malformed query, expected [END_OBJECT] but found [FIELD_NAME]","line":1,"col":53}],"type":"parsing_exception","reason":"[query_string] malformed query, expected [END_OBJECT] but found [FIELD_NAME]","line":1,"col":53},"status":400}
```

As far as I can see, all three variants work when changing the code in `ElasticsearchSearcher`:

```php
// old
//                $filter instanceof Condition\SearchCondition => $query['query_string']['query'] = $filter->query,
// new
                $filter instanceof Condition\SearchCondition => $query['bool']['must']['query_string']['query'] = $filter->query,
```

**Comment by alexander-schranz** [comment]
Hello @dhirtzbruch,

Thx for the report. So the issue is to use a `EqualCondition` and a `SearchCondition` together. Can you try to provide a failing test case before we have a deeper look how to integrate a fix here, so we first know which search engines fail in this case.

You could copy the following test https://github.com/schranz-search/schranz-search/blob/c91a3b7032785c91d541628714cd7ee2a75cb3bc/packages/seal/src/Testing/AbstractSearcherTestCase.php#L238 and add additional matching [SearchCondition](https://github.com/schranz-search/schranz-search/blob/c91a3b7032785c91d541628714cd7ee2a75cb3bc/packages/seal/src/Testing/AbstractSearcherTestCase.php#L128) to it.

**Comment by dhirtzbruch** [comment]
Of course, I'll check that tomorrow.

**Comment by alexander-schranz** [comment]
fixed via #314
